src/AeroSharp/DataAccess/KeyValueAccess/KeyValueStore.cs

```diff
@@ -166,9 +166,8 @@ private T GetValueToWrite<T>(Func<T> addValueFunc, Func<T, T> updateValueFunc, I
         {
             // Adding this function in as a helper method to AddOrUpdate
             // Intention is to make the code in the retry simple and readable
-
             T result;
-            if (value.First().Value is object)
+            if (value.First().Value is object && MapRecordsToTuple<T>(value, bin).First().Value is object)
```

Comment:
Since we're using the value of of the result in `MapRecordsToTuple<T>(value, bin).First().Value` I'd recommend storing it in a variable. 

`MapRecordsToTuple<T>` parses the blob, which consequently deserializes it into an object of type T.

In instances where the payload is also compressed, decompression would be happening too-- so we want to avoid unnecessary expense.